packages/format-library/src/language/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+import {
+	TextControl,
+	SelectControl,
+	Button,
+	Popover,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { applyFormat, removeFormat, useAnchor } from '@wordpress/rich-text';
+import { translation } from '@wordpress/icons';
+
+const name = 'core/language';
+const title = __( 'Language' );
+
+const Edit = ( props ) => {
+	const { contentRef, isActive, onChange, value } = props;
+	const anchorRef = useAnchor( {
+		editableContentElement: contentRef.current,
+		language,
+	} );
+
+	const [ lang, setLang ] = useState( '' );
+	const [ dir, setDir ] = useState( 'ltr' );
+
+	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
+	const togglePopover = () => {
+		setIsPopoverVisible( ( state ) => ! state );
+		setLang( '' );
+		setDir( 'ltr' );
+	};
+
+	return (
+		<>
+			<RichTextToolbarButton
```

Comment:
Should this be a radio button? Nothing is indicated to screen readers when a language tag is applied at this control level.

Reply:
Thank you for raising this Alex!
The was a missing `role="menuitemcheckbox"` there, I added it and it should be OK now 👍

Reply:
Good catch Alex ❤️ Yep, this should work like the Highlight, Inline code, and Strikethrough buttons. With a `role="menuitemcheckbox"` and an `aria-checked` true/false.